### PR TITLE
Assorted fixes

### PR DESCRIPTION
**Pull request by manast**
small misc. fixes found during 3.0 development

### CODE REVIEW

File: lib/queue.js [hunk 1]
```diff
@@ -82,7 +82,7 @@ var Queue = function Queue(name, url, opts){
   }
 
   if(_.isString(url)){
-    opts = _.defaults(redisOptsFromUrl, opts);
+    opts = {redis: _.defaults(redisOptsFromUrl(url), opts)};
```
Comment (bradvogel) [inline lib/queue.js]: I don't think this is right - it merges `opts` into the redis options, which is not what the user expects. I think it should be like https://github.com/OptimalBits/bull/pull/538/files#diff-b68fbb0f945cdf9cddec65ddf0fbe456R85 where we only override `opts.redis` and merge in the user-passed options.

Comment (bradvogel) [inline lib/queue.js]: I don't think this is right - it merges opts into the redis options, which is not what the user expects. I think it should be like https://github.com/OptimalBits/bull/pull/538/files#diff-b68fbb0f945cdf9cddec65ddf0fbe456R85 where we only override opts.redis and merge in the user-passed options.

Comment (manast) [inline lib/queue.js]: yes, that is completely wrong, I guess it was too late for coding when i did that :)

**Comment by bradvogel** [comment]
this lgtm, though the tests are failing in CI

**Comment by manast** [comment]
yes, it is strange, even without the change they fail I have to look into it.

**Comment by manast** [comment]
maybe not so strange, there are other changes that breaks the tests

**Comment by bradvogel** [comment]
this lgtm to be merged, if you can get tests to pass

**Comment by bradvogel** [comment]
v3.0.0 is so close! Hope to start using it this week and use it on one of our high-volume production services :)

**Pull request merged**